keps/sig-node/3008-cri-class-based-resources/README.md

```diff
@@ -0,0 +1,1280 @@
++    // Pod-level resources. Currently, requests and limits are not allowed
++    // to be specified for pods.
++    // +optional
++    Resources ResourceRequirements
+ }
+```
+
+In practice, the class resource information will be directly used in the CRI
+ContainerConfig (e.g.  CreateContainerRequest message). At this point, without
+resource discovery or access control kubelet does not do any validity checking
+of the values. Invalid class assignments will cause an error in the container
+runtime which causes the corresponding CRI RuntimeService request (e.g.
+RunPodSandbox or CreateContainer) to fail with an error.
+
+Input validation of classes very similar to labels is implemented: keys
+(`ClassResourceName`) and values must be non-empty, less than 64 characters
+long, must start and end with an alphanumeric character and may contain only
+alphanumeric characters, dashes, underscores or dots (`-`, `_` or `.`).
+Similar to labels, a namespace prefix (FQDN subdomain separated with a slash)
+in the key is allowed, similar to labels, e.g. `vendor/resource`.
+
+#### Resource status/capacity
+
+This KEP does not speak out anything about presenting the available resource
+types (or classes within) to the users.
+
+Some alternatives for presenting this information:
+
+1. Supplement `NodeStatus`
+
+   ```diff
+    // NodeStatus is information about the current status of a node.
+    type NodeStatus struct {
+            // Capacity represents the total resources of a node.
+            // More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#capacity
+            // +optional
+            Capacity ResourceList `json:"capacity,omitempty" protobuf:"bytes,1,rep,name=capacity,casttype=ResourceList,castkey=ResourceName"`
+            // Allocatable represents the resources of a node that are available for scheduling.
+            // Defaults to Capacity.
+            // +optional
+            Allocatable ResourceList `json:"allocatable,omitempty" protobuf:"bytes,2,rep,name=allocatable,casttype=ResourceList,castkey=ResourceName"`
+   +        // ResourceClasses lists the available
+   +        ClassResourdes []ClassResourceList
+   +
+   +type ClassResourceList {
+   +        // Name of the resource
+   +        Name ClassResourceName
+   +        // Classes available in the resource
+   +        Classes []string
+   +}
+   ```
+1. Separate API objects (e.g. something like `RuntimeClass`). Doesn't
+   necessarily that neatly align with two level hierarchy (resource name and a
+   set of classes within). Also, only best suited to homogenous clusters.
+
+#### Resource discovery
+
+Some possible alternatives.
+
+1. Reported by the container runtime. Container runtime is (or at least should
+   be) aware of all resource types and the classes within. It could advertise
+   the resources e.g. via either:
+
+   1. A separate gRPC endpoint or update `StatusResponse`
```

Comment:
as a note of preference, this seems more idiomatic for how the cri behaves today

Reply:
I can't see the note (to the point where I almost posted the exact same note as I did three months ago :) ). Maybe update the list to highlight the current direction, and add in the `alternatives` section below some alternatives to it?